pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -122,13 +127,23 @@ func New(ctx context.Context, plArgs runtime.Object, fh framework.Handle, fts fe
 		return &DynamicResources{}, nil
 	}
 
+	args, ok := plArgs.(*config.DynamicResourcesArgs)
+	if !ok {
+		return nil, fmt.Errorf("got args of type %T, want *DynamiccResourcesArgs", plArgs)
+	}
+	if err := validation.ValidateDynamicResourcesArgs(nil, args); err != nil {
+		return nil, err
+	}
```

Comment:
nit: so that the validation isn't triggered when the feature gate is off

```suggestion
	if fts.EnableDRASchedulerFilterTimeout {
		if err := validation.ValidateDynamicResourcesArgs(nil, args); err != nil {
			return nil, err
		}
	}
```

Reply:
Okay. I suppose the corresponding behavior in the apiserver is to drop disabled fields before they reach validation, i.e. invalid values are okay when the corresponding feature is off.

However, we only want to disable validation of that one field. Other fields might get added later, so we need to push the feature gates into `ValidateDynamicResourcesArgs` and do the if check there.

Reply:
Yes, but then I included that additional commit in the wrong branch. Thanks for double-checking! It's good to know that you are diligent about reviews despite the time constraints.

It's here now, see 8c13b2c99beb5fe5f46f18a3ae26ec141909b892.

Reply:
This turned out be more complicated: that commit broke `validatePluginConfig`, which called `ValidateDynamicResourcesArgs` through reflection => runtime error instead of compile error, so I hadn't noticed.

To solve this, I had to find a way to construct the struct from the `DefaultFeatureGate` also in that code. I ended up moving the construction next to the struct, which should also make future extensions more localized.

See the [updated commit](https://github.com/kubernetes/kubernetes/pull/132033/commits/b858402a370b531e7e689f5afe92e18a562dff80) for the complete solution.

Reply:
Thanks, yes, `features.NewFeatures` makes sense.

/approve
from sig-scheduling (non API area